frontend/src/components/AgreementCard.tsx

```diff
@@ -42,76 +44,110 @@ const AgreementCard: FC<IAgreementCardProps> = ({
+                <DropDownMenu
+                  mainIcon={<MoreHorizIcon />}
+                  name="agreement-menu"
+                  menuItems={[
+                    {
+                      text: t(isArchived ? StringBank.UNARCHIVE : StringBank.ARCHIVE),
+                      icon: <Inventory2OutlinedIcon />,
+                      action: () => archiveAgreement(id, !isArchived),
+                    },
+                    {
+                      text: 'Delete',
+                      icon: <DeleteOutlineIcon sx={{ color: 'red' }} />,
+                      action: () => {
+                        handleClickOpenDialog();
+                      },
+                    },
+                  ]}
+                />
+                <Typography
+                  fontWeight="bold"
+                  variant="body1"
+                  overflow="hidden"
+                  textOverflow="ellipsis"
+                  whiteSpace="nowrap"
+                  maxWidth="calc(100% - 40px)"
+                >
+                  {title}
+                </Typography>
+              </Stack>
+
+              <Typography variant="caption">
+                {t(StringBank.AGREEMENT_PARTICIPANTS, { count: participants })}
+              </Typography>
+              <Typography variant="body2" marginY={1} height="4.3em">
+                {truncateAfterWords(rationale ?? ' ', 12)}
+              </Typography>
+              <Typography variant="caption">
+                {t(StringBank.AGREEMENT_UPDATED_AT, {
+                  date: updatedAt.toLocaleString(navigator.language, {
+                    month: 'numeric',
+                    day: 'numeric',
+                    year: 'numeric',
+                    hour: 'numeric',
+                    minute: 'numeric',
+                  }),
+                })}
               </Typography>
             </Stack>
-            <Typography variant="caption">
-              {t(StringBank.AGREEMENT_PARTICIPANTS, { count: participants })}
-            </Typography>
-            <Typography variant="body2" marginY={1} height="4.3em">
-              {truncateAfterWords(rationale ?? ' ', 12)}
-            </Typography>
-            <Typography variant="caption">
-              {t(StringBank.AGREEMENT_UPDATED_AT, {
-                date: updatedAt.toLocaleString(navigator.language, {
-                  month: 'numeric',
-                  day: 'numeric',
-                  year: 'numeric',
-                  hour: 'numeric',
-                  minute: 'numeric',
-                }),
-              })}
-            </Typography>
-          </Stack>
-        </CardContent>
-      </Stack>
-    </MainCard>
+          </CardContent>
+        </Stack>
+      </MainCard>
+      <DialogEl
+        openDialogState={openDialogState}
+        title="Delete Agreement"
+        content="Please write the name of the agreement to delete it"
+        closeFunction={handleCloseDialog}
+        createFunction={onDeleteAgreement}
```

Comment:
@yishain11 `createFunction` prop is too specific and doesn't work so well here, maybe you have an idea for a more generic name?